Dynatrace/dynatrace-operator
# Add e2e test with cloudnative upgrade scenario

# Description

It covers operator upgrade in cloudNativeMode e2e test scenario:
- install latest DT Operator release from github [1]
- deploy sample apps
- upgrade Operator to snapshot version
- restart half of sample pods
- check if all sample apps are monitored

[1] Currently `releaseTag` to be checkouted is hardcoded as `v0.9.1`. We may want it configurable, e.g. via linker flags. To be discussed.

Logic change that may affect other tests:
- replace `kubernetes-all.yaml` usage with `kubernetes.yaml + kubernetes-csi.yaml` to reflect production environment

## How can this be tested?
`make test/e2e/cloudnative`

## Checklist
